File: length_of_last_word.py
class Solution:
    # @param s, a string
    # @return an integer
    def lengthOfLastWord(self, s):
        if s == '': return 0
        k = len(s)
        i = k-1
        while s[i] == ' ' and i >= 0:
            i -= 1
        le = 0
        while s[i] != ' ' and i >= 0:
            le += 1
            i -= 1
        return le

File: Leetcode_Java/LengthOfLastWord.java
public class LengthOfLastWord {
    public int lengthOfLastWord(String s) {
        if (s == null || s.length() == 0) {
            return 0;
        }
        int length = s.length();
        int i = s.length() - 1;
        while (i >= 0 && s.charAt(i) == ' ') {
            i--;
        }
        if (i == -1) {
            return 0;
        }
        int end = i;
        while (i >= 0 && s.charAt(i) != ' ') {
            i--;
        }
        return end - i;
    }
}